### ISSUE TITLE
Add support for choosing a user context to display the user interface in

### ISSUE THREAD

**Issue by kenkendk** [duplicati/duplicati#1739]
If Duplicati is running as a service, the user context is that of the service.
This means that the users home folders (music, pictures, etc) does not display correctly in the user interface.

It should be possible to choose what user context to use. It could be saved as a cookie, and possibly passed from the tray-icon when opening the UI so most users will not be bothered by it.

If the UI needs to display something that requires a user context (like a file picker) it should ask the user what context should be used. Perhaps this should be change-able on the settings page.

**Comment by kenkendk** [comment]
Some additional ideas are mentioned in #1768

**Comment by kenkendk** [comment]
Perhaps we should just require that the service has a user-context assigned (the user installing the program perhaps?).

That would not work well in a multi-user setup, but such a setup may not want the users to setup backups anyway.

**Comment by mnaiman** [comment]
I agree. I think it is good to provide special folders if program is installed under user account.
On the other hand if it is localsystem special folders could be hidden in web interface and problem is gone.
Or to enumerate all users on system and show their special folders.

I think, there will be 3 type of users
1)system admins (require localsystem) for ability to backup anything, or use on server
2)end users, but they are all running their PCs as "admin" account (defualt after instal) so they have access anywhere on computer otherwise
3)end users which have limited account - but therefore lacking ability to install any software including duplicati

**Comment by FlohEinstein** [comment]
Don't make it more complicated than it should be.
Start "services.msc", go to "Duplicati service".
Right-click, Properties, "Log On".
Enter username and password of the user account you want to use
Click ok.
First time you do this it will show you "User has been granted right 'Log On As A Service"
Voilà, you're in the right context.

**Comment by agrajaghh** [comment]
> Don't make it more complicated than it should be.
> Start "services.msc", go to "Duplicati service".
> Right-click, Properties, "Log On".
> Enter username and password of the user account you want to use
> Click ok.
> First time you do this it will show you "User has been granted right 'Log On As A Service"
> Voilà, you're in the right context.

is it possible to to this automatically with the installer? and maybe even switch the user later from within duplicati?

**Comment by emaktech** [comment]
I will add my $0.02 to this. I think the ideal installation scenario would be thus:

1) First screen, ask if you want to install in service mode or user mode with a radio button (default to user mode to keep it simple for most people).

2) User mode continues install as it is already.

3) Service mode completes install as a service (run Duplicati.WindowsService.exe install --webservice-interface=loopback --log-retention=3M) at the end of the install.

4) When installed in this mode, change the desktop icon to either just be a shortcut to the Duplicati localhost webpage.

5) If possible it would be great if the tray icon could appear in the user context and show you what state the service is in (green running, black idle, red error) but this is probably much less important.

Does this make sense? This would be great when setting up Duplicati in a managed environment as it would vastly simplify the install process. Maybe setup a small funding goal to get this done?

**Comment by emaktech** [comment]
Another comment as well. When Duplicati is set up to run as a service, when it is updated the following needs to be done to complete the update or it won't work:

1) Stop service
2) Run Duplicati.WindowsService.exe uninstall
3) Run Duplicati.WindowsService.exe install --webservice-interface=loopback --log-retention=3M
4) Start service

Also, it behaves much better if during install you set Duplicati to not start on sign-in.

**Comment by mnaiman** [comment]
@emaktech 
1)tray icon support for reading status of service is prepared in https://github.com/duplicati/duplicati/pull/2451
2)why after update this need to be done? Im not doing that and update of service is running fine
3)any parameters specification during install is not good idea (except parameters before config is loaded like --portable ...), rather configure at runtime how long you want to have logs kept and which interface to listen to.

**Comment by emaktech** [comment]
@mnaiman 
1) Good stuff on the tray icon, very cool. Thanks for the link.
2) I'm not sure why it's necessary, but I have had to do that on two different systems I just set up when installing the updates from within Duplicati. Otherwise, I was not able to connect from the browser, after following this process it works. When updating are you downloading and installing manually or doing it from within the web interface (download/activate)
3) I have no real preference on parameters. I just saw how to set up the process from @kenkendk in another thread and followed that process since. I'm sure you know better than I do here :)

**Comment by mnaiman** [comment]
@emaktech 
2) interesting - Im doing that from web UI - update and everytime it loaded new version fine and I was able to connect.
One difference maybe - Im running Duplicati as service but portable mode (not installed by setup, but installed via Duplicati.WindowsService.exe install --portable
Maybe that in your case update process has some file permissions problem (but its weird, because it runs under LOCALSYSTEM)
